### PR TITLE
Wizard recipe: MQLib-v0.1.0

### PR DESCRIPTION
**Pull request by pedromxavier**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: MQLib
* Version: v0.1.0

@staticfloat please review and merge.

### CODE REVIEW

File: M/MQLib/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("x86_64", "freebsd"; )
+]
```
Comment (giordano) [inline M/MQLib/build_tarballs.jl:39]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: M/MQLib/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+products = [
+    ExecutableProduct("MQLib", :MQLib)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"10.2.0")
```
Comment (giordano) [inline M/MQLib/build_tarballs.jl:52]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

File: M/MQLib/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/MQLib/
+
+make
+
+cp ${WORKSPACE}/srcdir/MQLib/bin/MQLib ${prefix}/MQLib
+cp ${WORKSPACE}/srcdir/MQLib/bin/MQLib.a ${prefix}/MQLib.a
```
Comment (giordano) [inline M/MQLib/build_tarballs.jl:20]:
```suggestion
make
install -Dvm 0755 bin/MQLib "${bindir}/MQLib${exeext}"
```

**Comment by pedromxavier** [comment]
@giordano, thanks for that! I think this project is not prepared to be compiled for windows and darwin systems.
I'm considering making a fork and rewrite some makefiles and headers (maybe aiming to compile a shared lib too...)

**Comment by giordano** [comment]
You don't really need a fork, you can apply the following patches.  For Windows (see https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170):
```diff
diff --git a/src/heuristics/maxcut/burer2002.cpp b/src/heuristics/maxcut/burer2002.cpp
index b54e793..edeeebd 100644
--- a/src/heuristics/maxcut/burer2002.cpp
+++ b/src/heuristics/maxcut/burer2002.cpp
@@ -1,3 +1,6 @@
+#ifdef _WIN32
+#define _USE_MATH_DEFINES
+#endif
 #include <math.h>
 #include <string.h>
 #include <algorithm>
```
for macOS (but really for any platforms, this header is really needed and not including it is simply a mistake):
```
diff --git a/src/heuristics/qubo/glover2010.cpp b/src/heuristics/qubo/glover2010.cpp
index e926b6e..bc4ea48 100644
--- a/src/heuristics/qubo/glover2010.cpp
+++ b/src/heuristics/qubo/glover2010.cpp
@@ -3,6 +3,7 @@
 #include <algorithm>
 #include <iostream>
 #include <vector>
+#include <limits>
 #include "heuristics/qubo/glover2010.h"
 #include "util/random.h"
 
diff --git a/src/heuristics/qubo/katayama2000.cpp b/src/heuristics/qubo/katayama2000.cpp
index 9a3c909..04595d5 100644
--- a/src/heuristics/qubo/katayama2000.cpp
+++ b/src/heuristics/qubo/katayama2000.cpp
@@ -1,5 +1,6 @@
 #include <algorithm>
 #include <iostream>
+#include <limits>
 #include "heuristics/qubo/katayama2000.h"
 #include "util/random.h"
 
diff --git a/src/heuristics/qubo/merz2002.cpp b/src/heuristics/qubo/merz2002.cpp
index 2c46d63..052bce0 100644
--- a/src/heuristics/qubo/merz2002.cpp
+++ b/src/heuristics/qubo/merz2002.cpp
@@ -1,5 +1,6 @@
 #include <algorithm>
 #include <iostream>
+#include <limits>
 #include "heuristics/qubo/merz2002.h"
 #include "util/random.h"
 
diff --git a/src/heuristics/qubo/merz2004.cpp b/src/heuristics/qubo/merz2004.cpp
index 40f2bb2..f680ddf 100644
--- a/src/heuristics/qubo/merz2004.cpp
+++ b/src/heuristics/qubo/merz2004.cpp
@@ -1,5 +1,6 @@
 #include <algorithm>
 #include <iostream>
+#include <limits>
 #include "heuristics/qubo/merz2004.h"
 #include "util/random.h"
 
```

**Comment by giordano** [comment]
Thanks for your contribution!

**Pull request merged**